### ISSUE TITLE
Unable to initialize GLEW!

### ISSUE THREAD

**Issue by eternal-sorrow** [flathub/io.github.endless_sky.endless_sky#1]
I get this error in terminal when I try to start Endless Sky.

**Comment by eternal-sorrow** [comment]
Running Endless Sky from Steam (which is also installed via Flatpak) works smoothly.

**Comment by corecontingency** [comment]
What is the results of `flatpak list`? What distro are you using? I'd like to try to reproduce this.

You are starting the flatpak'd Endless Sky with `flatpak run io.github.endless_sky.endless_sky`, correct?

**Comment by corecontingency** [comment]
Can you post the full logs from running `flatpak run io.github.endless_sky.endless_sky` in a terminal? Thanks.

**Comment by eternal-sorrow** [comment]
>What is the results of flatpak list?

I've already uninstalled the endless sky, but it didn't pull any additional dependencies.

```
>flatpak list
Flatseal                                            com.github.tchx84.Flatseal                               1.6.0               stable            user
Steam                                               com.valvesoftware.Steam                                  1.0.0.64            stable            user
Code - OSS                                          com.visualstudio.code-oss                                1.47.1              stable            user
Freedesktop Platform                                org.freedesktop.Platform                                 19.08.12            19.08             user
i386                                                org.freedesktop.Platform.Compat.i386                                         19.08             user
default                                             org.freedesktop.Platform.GL.default                                          19.08             user
nvidia-390-132                                      org.freedesktop.Platform.GL.nvidia-390-132                                   1.4               user
default                                             org.freedesktop.Platform.GL32.default                                        19.08             user
nvidia-390-132                                      org.freedesktop.Platform.GL32.nvidia-390-132                                 1.4               user
Intel                                               org.freedesktop.Platform.VAAPI.Intel                                         19.08             user
ffmpeg-full                                         org.freedesktop.Platform.ffmpeg-full                                         19.08             user
openh264                                            org.freedesktop.Platform.openh264                        2.1.0               2.0               user
Freedesktop SDK                                     org.freedesktop.Sdk                                      19.08.12            19.08             user
GNOME Application Platform version 3.36             org.gnome.Platform                                                           3.36              user
KDE Application Platform                            org.kde.Platform                                                             5.14              user
Telegram Desktop                                    org.telegram.desktop                                     2.1.13              stable            user
```

>What distro are you using?

Gentoo

>You are starting the flatpak'd Endless Sky with flatpak run io.github.endless_sky.endless_sky, correct?

Yes.

>Also, can you post the full logs from running flatpak run io.github.endless_sky.endless_sky in a terminal? Thanks.

```
Unable to initialize GLEW!
```

That is the full log. It didn't output anything else. Even after I tried adding various command line options like `-d` or `-v`.

**Comment by corecontingency** [comment]
Huh. GLEW is already packaged in with this flatpak. From my (limited) understanding, it lets Endless Sky know what OpenGL extensions your computer supports.

Just guessing here, but possibly something is broken with your NVIDIA driver, so when GLEW tries to query it about OpenGL support, GLEW gets confused and exits?

Sorry that I am not able to help more.

**Comment by eternal-sorrow** [comment]
`org.freedesktop.Platform.GL.nvidia-390-132` is installed to be used when I run separate Xorg with my hybrid NVIDIA GPU (with nvidia-xrun). I thought it's not used by flatpak when I start the application using my Intel GPU, and looks like I was right. I uninstalled it to be sure it does not mess up with GLEW and reinstalled Endless Sky, but result was the same as before.

**Comment by eternal-sorrow** [comment]
Okay, I installed thist game as a native application (not in flatpak) and got the same error, so this is clearly not a flatpak issue.

**Comment by eternal-sorrow** [comment]
Okay, is it possible to patch the game sources while building the flatpak? I found the small patch that fixes this error on wayland.

```diff
--- a/source/main.cpp	2020-08-09 14:15:36.580904801 +1000
+++ b/source/main.cpp	2020-08-09 14:20:58.508311005 +1000
@@ -152,7 +152,8 @@
 		// Initialize GLEW.
 #ifndef __APPLE__
 		glewExperimental = GL_TRUE;
-		if(glewInit() != GLEW_OK)
+		
+		if(GLenum err = glewInit(); GLEW_OK != err && err != GLEW_ERROR_NO_GLX_DISPLAY)
 			return DoError("Unable to initialize GLEW!", window, context);
 #endif
 		
```
This appears to be the GLEW issue https://github.com/nigels-com/glew/issues/273

**Comment by Pointedstick** [comment]
You should submit that fix upstream: https://github.com/endless-sky/endless-sky/pulls. I'll review it there. :)

**Comment by eternal-sorrow** [comment]
@Pointedstick done https://github.com/endless-sky/endless-sky/pull/5217

**Comment by StevenH928** [comment]
Had the same issue on Fedora 36 - Gnome 42.4 - Wayland
RPM pkg 
NVIDIA not the issue as those drivers are not present.

**Comment by Pointedstick** [comment]
Should be fixed in the 0.9.15 release now.
